support-frontend/app/views/subscriptionCheckout.scala.html

```diff
@@ -40,11 +40,25 @@
 
       window.guardian.orderIsAGift = @orderIsAGift
 
-      window.guardian.user = {
-        firstName: "@user.map(_.privateFields.firstName).getOrElse("")",
-        lastName: "@user.map(_.privateFields.secondName).getOrElse("")",
-        email: "@user.map(_.primaryEmailAddress).getOrElse("")"
-      };
+      @idUser.map { user =>
+        window.guardian.user = {
+          id: "@user.id",
+          email: "@user.primaryEmailAddress",
+          @user.publicFields.displayName.map { displayName =>
+            displayName: "@displayName",
+          }
+          @for(firstName <- user.privateFields.firstName; lastName <- user.privateFields.secondName) {
+            firstName: "@firstName",
+            lastName: "@lastName",
```

Comment:
Given a user who only has one name (first or last) set, will this leave out the name from the response? If so, does that matter?

Reply:
It shouldn't matter, the user can enter their name manually into the form.